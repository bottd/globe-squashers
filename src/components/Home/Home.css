@import url('https://fonts.googleapis.com/css?family=Josefin+Sans');

.Home {
  position: relative;
}

.NavLink-create {
  margin: 0 auto;
  color: #fff;
  display: flex;
  flex-direction: column;
  position: fixed;
  text-decoration: none;
  top: 0;
  right: 0;
  z-index: 999;
}

.Home-title-page {
  background-image: url('https://cors-anywhere.herokuapp.com/http://imgsrc.hubblesite.org/hvi/uploads/image_file/image_attachment/28735/xlarge_web.jpg');
  background-repeat: no-repeat;
  background-size: cover;
  height: 100vh;
  width: 100vw;
}

.title-container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  height: 100vh;
  width: 100vw;
}

.title-box {
  background-color: rgba(0, 0, 0, 0.7);
  margin: 0 auto;
  padding: 20px;
  width: 80%;
}

.Home-title {
  border-bottom: 7px solid #fff;
  color: #fff;
  margin: 0 auto;
  font-size: 8em;
  font-family: 'Josefin Sans', sans-serif;
}

.Home-subheader {
  color: #fff;
  margin: 15px auto;
  font-size: 3em;
  font-family: 'Josefin Sans', sans-serif;
}

.button-header {
  margin: 0 auto;
  color: #fff;
  font-size: 3em;
  font-family: 'Josefin Sans', sans-serif;
}

.button-wrapper {
  position: absolute;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  bottom: 100vh;
  width: 100%;
  margin: 0 auto;
}
.scroll-to-info {
  border: none;
  background: transparent;
  width: 15%;
}

.scroll-to-info:focus {
  outline: 0;
}

i {
  height: 65px;
  color: #fff;
  font-size: 80px;
}
