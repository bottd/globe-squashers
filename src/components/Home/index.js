import React from 'react';
import {NavLink} from 'react-router-dom';
import InfoPage from './components/InfoPage';
import './Home.css';

export default function Home() {
  const smoothScroll = () => {
    window.scrollTo({
      behavior: 'smooth',
      top: document.body.scrollHeight,
    });
  };

  return (
    <div className="Home">
      <div className="Home-title-page">
        <div className="title-container">
          <div className="title-box">
            <h1 className="Home-title">GLOBE SQUASHERS</h1>
            <p className="Home-subheader">
              HELP PROVE THE HUBBLE TELESCOPE IS A LIE
            </p>
          </div>
          <div className="button-wrapper">
            <button className="scroll-to-info" onClick={smoothScroll}>
              <p className="button-header">MORE INFO</p>
              <i className="fas fa-caret-down" />
            </button>
          </div>
        </div>
      </div>
      <InfoPage />
    </div>
  );
}
