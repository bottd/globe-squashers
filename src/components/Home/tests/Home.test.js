import React from 'react';
import {shallow} from 'enzyme';
import Home from '../';

describe('Home component', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<Home />);
  });

  it('Should render like snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
