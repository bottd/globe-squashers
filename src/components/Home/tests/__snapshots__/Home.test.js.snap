// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component Should render like snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="Home-title-page"
        >
          <div
            className="title-container"
          >
            <div
              className="title-box"
            >
              <h1
                className="Home-title"
              >
                GLOBE SQUASHERS
              </h1>
              <p
                className="Home-subheader"
              >
                HELP PROVE THE HUBBLE TELESCOPE IS A LIE
              </p>
            </div>
            <div
              className="button-wrapper"
            >
              <button
                className="scroll-to-info"
                onClick={[Function]}
              >
                <p
                  className="button-header"
                >
                  MORE INFO
                </p>
                <i
                  className="fas fa-caret-down"
                />
              </button>
            </div>
          </div>
        </div>,
        <InfoPage />,
      ],
      "className": "Home",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="title-container"
          >
            <div
              className="title-box"
            >
              <h1
                className="Home-title"
              >
                GLOBE SQUASHERS
              </h1>
              <p
                className="Home-subheader"
              >
                HELP PROVE THE HUBBLE TELESCOPE IS A LIE
              </p>
            </div>
            <div
              className="button-wrapper"
            >
              <button
                className="scroll-to-info"
                onClick={[Function]}
              >
                <p
                  className="button-header"
                >
                  MORE INFO
                </p>
                <i
                  className="fas fa-caret-down"
                />
              </button>
            </div>
          </div>,
          "className": "Home-title-page",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="title-box"
              >
                <h1
                  className="Home-title"
                >
                  GLOBE SQUASHERS
                </h1>
                <p
                  className="Home-subheader"
                >
                  HELP PROVE THE HUBBLE TELESCOPE IS A LIE
                </p>
              </div>,
              <div
                className="button-wrapper"
              >
                <button
                  className="scroll-to-info"
                  onClick={[Function]}
                >
                  <p
                    className="button-header"
                  >
                    MORE INFO
                  </p>
                  <i
                    className="fas fa-caret-down"
                  />
                </button>
              </div>,
            ],
            "className": "title-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  <h1
                    className="Home-title"
                  >
                    GLOBE SQUASHERS
                  </h1>,
                  <p
                    className="Home-subheader"
                  >
                    HELP PROVE THE HUBBLE TELESCOPE IS A LIE
                  </p>,
                ],
                "className": "title-box",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "GLOBE SQUASHERS",
                    "className": "Home-title",
                  },
                  "ref": null,
                  "rendered": "GLOBE SQUASHERS",
                  "type": "h1",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "HELP PROVE THE HUBBLE TELESCOPE IS A LIE",
                    "className": "Home-subheader",
                  },
                  "ref": null,
                  "rendered": "HELP PROVE THE HUBBLE TELESCOPE IS A LIE",
                  "type": "p",
                },
              ],
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <button
                  className="scroll-to-info"
                  onClick={[Function]}
                >
                  <p
                    className="button-header"
                  >
                    MORE INFO
                  </p>
                  <i
                    className="fas fa-caret-down"
                  />
                </button>,
                "className": "button-wrapper",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <p
                      className="button-header"
                    >
                      MORE INFO
                    </p>,
                    <i
                      className="fas fa-caret-down"
                    />,
                  ],
                  "className": "scroll-to-info",
                  "onClick": [Function],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "MORE INFO",
                      "className": "button-header",
                    },
                    "ref": null,
                    "rendered": "MORE INFO",
                    "type": "p",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "className": "fas fa-caret-down",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": "i",
                  },
                ],
                "type": "button",
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="Home-title-page"
          >
            <div
              className="title-container"
            >
              <div
                className="title-box"
              >
                <h1
                  className="Home-title"
                >
                  GLOBE SQUASHERS
                </h1>
                <p
                  className="Home-subheader"
                >
                  HELP PROVE THE HUBBLE TELESCOPE IS A LIE
                </p>
              </div>
              <div
                className="button-wrapper"
              >
                <button
                  className="scroll-to-info"
                  onClick={[Function]}
                >
                  <p
                    className="button-header"
                  >
                    MORE INFO
                  </p>
                  <i
                    className="fas fa-caret-down"
                  />
                </button>
              </div>
            </div>
          </div>,
          <InfoPage />,
        ],
        "className": "Home",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="title-container"
            >
              <div
                className="title-box"
              >
                <h1
                  className="Home-title"
                >
                  GLOBE SQUASHERS
                </h1>
                <p
                  className="Home-subheader"
                >
                  HELP PROVE THE HUBBLE TELESCOPE IS A LIE
                </p>
              </div>
              <div
                className="button-wrapper"
              >
                <button
                  className="scroll-to-info"
                  onClick={[Function]}
                >
                  <p
                    className="button-header"
                  >
                    MORE INFO
                  </p>
                  <i
                    className="fas fa-caret-down"
                  />
                </button>
              </div>
            </div>,
            "className": "Home-title-page",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="title-box"
                >
                  <h1
                    className="Home-title"
                  >
                    GLOBE SQUASHERS
                  </h1>
                  <p
                    className="Home-subheader"
                  >
                    HELP PROVE THE HUBBLE TELESCOPE IS A LIE
                  </p>
                </div>,
                <div
                  className="button-wrapper"
                >
                  <button
                    className="scroll-to-info"
                    onClick={[Function]}
                  >
                    <p
                      className="button-header"
                    >
                      MORE INFO
                    </p>
                    <i
                      className="fas fa-caret-down"
                    />
                  </button>
                </div>,
              ],
              "className": "title-container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    <h1
                      className="Home-title"
                    >
                      GLOBE SQUASHERS
                    </h1>,
                    <p
                      className="Home-subheader"
                    >
                      HELP PROVE THE HUBBLE TELESCOPE IS A LIE
                    </p>,
                  ],
                  "className": "title-box",
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "GLOBE SQUASHERS",
                      "className": "Home-title",
                    },
                    "ref": null,
                    "rendered": "GLOBE SQUASHERS",
                    "type": "h1",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "HELP PROVE THE HUBBLE TELESCOPE IS A LIE",
                      "className": "Home-subheader",
                    },
                    "ref": null,
                    "rendered": "HELP PROVE THE HUBBLE TELESCOPE IS A LIE",
                    "type": "p",
                  },
                ],
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <button
                    className="scroll-to-info"
                    onClick={[Function]}
                  >
                    <p
                      className="button-header"
                    >
                      MORE INFO
                    </p>
                    <i
                      className="fas fa-caret-down"
                    />
                  </button>,
                  "className": "button-wrapper",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": Array [
                      <p
                        className="button-header"
                      >
                        MORE INFO
                      </p>,
                      <i
                        className="fas fa-caret-down"
                      />,
                    ],
                    "className": "scroll-to-info",
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": Array [
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "children": "MORE INFO",
                        "className": "button-header",
                      },
                      "ref": null,
                      "rendered": "MORE INFO",
                      "type": "p",
                    },
                    Object {
                      "instance": null,
                      "key": undefined,
                      "nodeType": "host",
                      "props": Object {
                        "className": "fas fa-caret-down",
                      },
                      "ref": null,
                      "rendered": null,
                      "type": "i",
                    },
                  ],
                  "type": "button",
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
